bootstrap/util/configowner_test.go

```diff
@@ -20,113 +20,134 @@ import (
+	"sigs.k8s.io/controller-runtime/pkg/client/fake"
 )
 
-func TestGetConfigOwnerSuccess(t *testing.T) {
+func TestGetConfigOwner(t *testing.T) {
+	g := NewGomegaWithT(t)
+
 	scheme := runtime.NewScheme()
-	if err := clusterv1.AddToScheme(scheme); err != nil {
-		t.Fatal("failed to register Cluster API objects to scheme")
-	}
+	g.Expect(clusterv1.AddToScheme(scheme)).To(Succeed())
 
-	myMachine := &clusterv1.Machine{
-		ObjectMeta: metav1.ObjectMeta{
-			Name:      "my-machine",
-			Namespace: "my-ns",
-			Labels: map[string]string{
-				clusterv1.MachineControlPlaneLabelName: "",
+	t.Run("should get the owner when present (Machine)", func(t *testing.T) {
+		myMachine := &clusterv1.Machine{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "my-machine",
+				Namespace: "my-ns",
+				Labels: map[string]string{
+					clusterv1.MachineControlPlaneLabelName: "",
+				},
 			},
-		},
-		Spec: clusterv1.MachineSpec{
-			ClusterName: "my-cluster",
-			Bootstrap: clusterv1.Bootstrap{
-				DataSecretName: pointer.StringPtr("my-data-secret"),
+			Spec: clusterv1.MachineSpec{
+				ClusterName: "my-cluster",
+				Bootstrap: clusterv1.Bootstrap{
+					DataSecretName: pointer.StringPtr("my-data-secret"),
+				},
 			},
-		},
-		Status: clusterv1.MachineStatus{
-			InfrastructureReady: true,
-		},
-	}
+			Status: clusterv1.MachineStatus{
+				InfrastructureReady: true,
+			},
+		}
 
-	c := fake.NewFakeClientWithScheme(scheme, myMachine)
-	objm := metav1.ObjectMeta{
-		OwnerReferences: []metav1.OwnerReference{
-			{
-				Kind:       "Machine",
-				APIVersion: clusterv1.GroupVersion.String(),
-				Name:       "my-machine",
+		c := fake.NewFakeClientWithScheme(scheme, myMachine)
+		obj := &bootstrapv1.KubeadmConfig{
+			ObjectMeta: metav1.ObjectMeta{
+				OwnerReferences: []metav1.OwnerReference{
+					{
+						Kind:       "Machine",
+						APIVersion: clusterv1.GroupVersion.String(),
+						Name:       "my-machine",
+					},
+				},
+				Namespace: "my-ns",
+				Name:      "my-resource-owned-by-machine",
 			},
-		},
-		Namespace: "my-ns",
-		Name:      "my-resource-owned-by-machine",
-	}
-	configOwner, err := GetConfigOwner(context.TODO(), c, objm)
-	if err != nil {
-		t.Fatalf("did not expect an error but found one: %v", err)
-	}
-	if configOwner == nil {
-		t.Fatal("expected a configOwner but got nil")
-	}
-	if configOwner.ClusterName() != "my-cluster" {
-		t.Fatalf("did not expect ClusterName: %q", configOwner.ClusterName())
-	}
-	if !configOwner.IsInfrastructureReady() {
-		t.Fatalf("did not expect InfrastructureReady: %v", configOwner.IsInfrastructureReady())
-	}
-	if configOwner.DataSecretName() == nil {
-		t.Fatalf("did not expect DataSecretName: %v", configOwner.DataSecretName())
-	}
-	if !configOwner.IsControlPlaneMachine() {
-		t.Fatalf("did not expect IsControlPlane: %v", configOwner.IsControlPlaneMachine())
-	}
-}
+		}
+		configOwner, err := GetConfigOwner(context.TODO(), c, obj)
+		g.Expect(err).ToNot(HaveOccurred())
```

Comment:
nit: `NotTo` would read better here than `ToNot`
```suggestion
		g.Expect(err).NotTo(HaveOccurred())
```